src/tunnel.cc

```diff
@@ -104,8 +104,10 @@ class TunnelStateData: public PeerSelectionInitiator
         if (http.valid() && http->getConn() && http->getConn()->serverBump() && http->getConn()->serverBump()->at(XactionStep::tlsBump2, XactionStep::tlsBump3))
             return false;
 #endif
-        return !(request != NULL &&
-                 (request->flags.interceptTproxy || request->flags.intercepted));
+        // TODO: add a boolean ClientHttpRequest::faked field to mark
+        // faked CONNECT requests instead of trying to guess whether a request was
+        // faked based on its port configuration and current state.
```

Comment:
The _current state_ part is irrelevant to this return value, right?

```suggestion
        // faked based on its port configuration.
```